Source/FeatureRegistry.swift

```diff
@@ -169,27 +202,18 @@ extension LabeledGroupItem { /* Collection Support */
 
 extension FeatureRegistry {
 
-    /// Returns the label values for the enabled features in the provided registry.
+    /// Returns the names for the enabled features in the provided registry.
+    /// If a feature is embedded in a Freature Group, the group name is prepended
+    /// and formatted like: "feature_group → feature_name"
     /// Primarily used for debugging.
     /// - Parameter featureRegistry: The feature registry that should be used to find enabled features
     @objc public static func enabledFeatures(in featureRegistry: FeatureRegistry) -> [String] {
-        return featureRegistry.features.reduce(into: Array<LabeledFeatureItem>()) { (result, labeledItem) in
-            if let labeledFeature = labeledItem as? LabeledFeatureItem {
-                result.append(labeledFeature)
-            } else if let labeledGroup = labeledItem as? LabeledGroupItem {
-                addFeatures(from: labeledGroup, toResult: &result)
-            }
-        }.filter { $0.feature.enabled }.map { $0.label }
-    }
 
-    /// Recursive helper function that traverses a FeatureGroup and adds LabeledFeatureItem to the provided `results` array.
-    private static func addFeatures(from featureGroup: LabeledGroupItem, toResult result: inout [LabeledFeatureItem]) {
-        featureGroup.forEach {
-            if let labeledFeature = $0 as? LabeledFeatureItem {
-                result.append(labeledFeature)
-            } else if let groupItem = $0 as? LabeledGroupItem {
-                addFeatures(from: groupItem, toResult: &result)
-            }
+        return featureRegistry.features.depthFirstMap( resultBuilder: { (groupStack, feature) -> String in
+            let mergedString = groupStack.map { $0.label.unCamelCased }.joined(separator: " → ")
+            return mergedString.isEmpty ? feature.label.unCamelCased : mergedString + " → \(feature.label.unCamelCased)"
```

Comment:
Alternative: `return (mergedString.isEmpty ? "" : mergedString) + " → \(feature.label.unCamelCased)"`

Reply:
Wouldn't that result in dropping the top level features?

Reply:
Doh.